pandas/core/groupby/ops.py

```diff
@@ -526,7 +526,13 @@ def _cython_operation(self, kind, values, how, axis, min_count=-1, **kwargs):
             func = self._get_cython_function(kind, how, values, is_numeric)
         except NotImplementedError:
             if is_numeric:
-                values = ensure_float64(values)
+                try:
+                    values = ensure_float64(values)
+                except TypeError:
+                    if lib.infer_dtype(values, skipna=False) == "complex":
```

Comment:
future PR should push this to pandas/core/dtypes/cast.py and just call here (maybe make ensure_float64_or_complex)

Reply:
yah.  i think there was some discussion of a one-pass variant of lib.infer_dtype that would go well with that

Reply:
great